cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/table/field/FieldValidator.java

```diff
@@ -91,4 +93,28 @@ public void validatePrimaryKeys(Collection<Field<?>> keys, boolean allowPrefix)
       i++;
     }
   }
+
+  /**
+   * Validate if the given keys are partial primary keys.
+   *
+   * @param keys the keys to validate
+   * @throws InvalidFieldException if the given keys have extra key which is not in primary key.
+   */
+  public void validatePartialPrimaryKeys(Collection<Field<?>> keys) throws InvalidFieldException {
+    Set<String> primaryKeys = new HashSet<>(tableSchema.getPrimaryKeys());
+    if (keys.size() > primaryKeys.size()) {
+      throw new InvalidFieldException(
+        tableSchema.getTableId(), keys,
+        String.format("Given keys %s contain more fields than the primary keys %s", keys, primaryKeys));
+    }
+
+    for (Field<?> key : keys) {
+      validateField(key);
+      if (!primaryKeys.contains(key.getName())) {
+        throw new InvalidFieldException(
+          tableSchema.getTableId(), keys,
+          String.format("Given keys %s are not partial primary keys %s", keys, primaryKeys));
```

Comment:
Can we make the exception message better to tell which key is not part of primary key?

Reply:
Sure, will update

Reply:
updated